dimager/templates/profile.html

```diff
@@ -0,0 +1,44 @@
+{% extends 'base.html' %}
+{% load staticfiles %}
+{% load thumbnail %}
+{% block content %}
+
+
+<div id='profile_content'>
+    <h1>Welcome, {% if user.first_name %}{{user.first_name}}{% else %}{{user}}{% endif %}!</h1>
```

Comment:
It'd be much more efficient if your user's profile had an `__str__`  method that returned an appropriate value to represent the user, regardless of whether they had set a `first_name`.  Then you could write something simple, like:

```jinja
<h1>Welcome, {{ user.profile }}</h1>
```